lib/src/metta/text.rs

```diff
@@ -50,6 +50,26 @@ impl<'a> SExprParser<'a> {
     pub fn parse(&mut self, tokenizer: &Tokenizer) -> Option<Atom> {
         while let Some(c) = self.it.peek() {
             match c {
+                ';' => {
+                    let c1 = c.clone();
+                    // ; = skip line ;; = skip sexpr
+                    self.it.next();
+                    let next = self.it.peek();
+                    match next {
+                        Some(n) => {
+                            if *n == c1 {
+                                // found ;; parse sexpr and skip it
+                                self.skip_sexpr(tokenizer);
+                            } else {
+                                // it's just ; iterate until \n
+                                self.skip_line()
```

Comment:
Here looks like token after `;` if it is not `;` again is skipped inside `self.skip_line()`. On `;\n` sequence it will skip the next line. Taking it into account I would propose removing `self.it.next();` from code of `skip_sexpr` and `skip_line`. And add it separately into the first branch of the `if`.

Making code more compact the final version:
```rust
';' => {
  self.it.next();
  match self.it.peek() {
    Some(c) => {
      if c == ';' {
        self.it.next();
        self.skip_sexpr();
      } else {
        self.skip_line();
      }
    None => return None;
  }
},

Reply:
yes, i added test to check if it's handled correctly